### ISSUE TITLE
postfix::virtual does not allow multiple destinations

### ISSUE THREAD

**Issue by kschu91** [camptocamp/puppet-postfix#164]
The [postfix documentation](http://www.postfix.org/virtual.5.html) allows defining multiple destinations via comma separated list:

```puppet
@example.com user1@foo.com, user2@bar.com
```
But with postfix::virtual this is not possible since the augeas pattern `/[*+.0-9=@-Z\\a-z-]+/` does not allow this.

Trying...
```puppet
postfix::virtual { '@example.com':
  ensure      => present,
  destination => 'user1@foo.com, user2@foo.com',
}
```
... produces an `Failed to match` error.

**Comment by pillarsdotnet** [comment]
@kschu91 -- If you can write a spec test that fails, I'll write the code that makes it pass.

**Comment by lynxnot** [comment]
I've also stumbled upon this problem with multiple destinations. 
We also use a redirection to /dev/null as [explained here](https://www.serverwatch.com/columns/article.php/3844371/Forwarding-a-Postfix-Virtual-Alias-to-devnull.htm)

For this to work I had to modify the augeas lens as follow:

```
(* View: word *)
let word = store /[A-Za-z0-9@\*.+=, \/_-]+/
```
thus adding a comma, a space and a forward slash. This is probably semantically wrong from the augeas lens perspective, but it does the trick. The file i had to modify is the one provided by augeas-libs package and not the one provided by this repo.

**Comment by comport3** [comment]
This code should reproduce the problem:
`node default {
  include postfix
  postfix::hash { '/etc/postfix/virtual': ensure => present}
  postfix::config { 'virtual_alias_maps':
    ensure  => present,
    value   => 'hash:/etc/postfix/virtual',
  }
  postfix::virtual { 'user@example.com' :
      ensure      => present,
      destination => 'user@example.net, user@example.org'
  }
}`

We have developed a workaround.

Edit line 35 of <Puppet Root>/share/augeas/lenses/dist/postfix_virtual.aug on the client (not server) and change it to:
`let word = store /[A-Za-z0-9@\*.+?\,?\ A-Za-z0-9@\*.+=_-]+/`

Run Puppet once, it should apply the config successfully.

Then revert the above change and it will stay in place perpetually.

Failure to revert the change will result in the multi-user config being added on each Puppet run.

**Comment by raphink** [comment]
The Augeas lens has always supported multiple destinations, but the Puppet code is not making use of it currently. I'll write a patch to accept an array of destinations.

**Comment by raphink** [comment]
Multiple destinations were implemented in #223. You're welcome to test it.
